src/eager.js

```diff
@@ -6,8 +6,14 @@ import Helpers from './helpers';
 import Promise from './base/promise';
 import EagerBase from './base/eager';
 
-const getAttributeUnique = (models, attribute) =>
-  _.uniq(_.map(models, m => m.get(attribute)));
+const getAttributeUnique = (models, attribute) => {
+  return _.uniq(_.map(models, m => {
+    // use formatted attributes so that morphKey and foreignKey will match
+    // attribute keys
+    const attributes = m.format(_.clone(m.attributes));
```

Comment:
[`_.clone(...)`](https://lodash.com/docs#clone) is a bit "heavy load" operation. Actually we would only need one key. Use [`_.pick(...)`](https://lodash.com/docs#pick) instead for specific key.